packages/service/src/widget/mypage/header/filter/category.tsx

```
@@ -0,0 +1,138 @@
+  onCategoryToggle,
+  selected,
+}: {
+  group: string;
+  onCategoryToggle: (category: string) => () => void;
+  selected: string[];
+}) {
+  const validatedGroup = isValidGroupName(group) ? group : '전체';
+  const selectedCategory =
+    validatedGroup === '전체'
+      ? categoryList.reduce<CategoryItemType[]>((acc, { category }) => [...acc, ...category], [])
+      : categoryList.find((item) => item.title === validatedGroup)?.category || [];
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)